vars/containerExecuteInsidePod.groovy

```diff
@@ -0,0 +1,90 @@
+import com.sap.piper.ConfigurationHelper
+import com.sap.piper.SysEnv
+import groovy.transform.Field
+
+@Field def STEP_NAME = 'containerExecuteInsidePod'
+
+@Field Set GENERAL_CONFIG_KEYS = ['jenkinsKubernetes']
+
+@Field Set PARAMETER_KEYS = ['dockerWorkspace',
+                             'dockerEnvVars',
+                             'containerMap']
+
+@Field Set STEP_CONFIG_KEYS = PARAMETER_KEYS
+
+void call(Map parameters = [:], body) {
+    def uniqueId = UUID.randomUUID().toString()
+
+    handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters) {
+
+        final script = parameters.script
+        if (script == null)
+            script = [commonPipelineEnvironment: commonPipelineEnvironment]
+
+        Map config = ConfigurationHelper
+            .loadStepDefaults(this)
+            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
+            .mixinStageConfig(script.commonPipelineEnvironment, parameters.stageName?:env.STAGE_NAME, STEP_CONFIG_KEYS)
+            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS)
+            .mixin(parameters, PARAMETER_KEYS)
+            .withMandatoryProperty('containerMap')
```

Comment:
Why is it mandatory?

Reply:
I could imagine that you just execute something in a pod, e.g. the checkout, without having additional tools installed.